hera_sim/tests/test_simulator.py

```diff
@@ -155,3 +157,139 @@ def test_adding_vis_but_also_returning():
 
     vis = sim.add_foregrounds("diffuse_foreground", Tsky_mdl=HERA_Tsky_mdl['xx'], ret_vis=True)
     np.testing.assert_array_almost_equal(vis, sim.data.data_array, decimal=5)
+
+
+if sys.version_info.major < 3 or \
+   sys.version_info.major > 3 and sys.version_info.minor < 4:
+    @raises(VersionError)
+    def test_run_sim():
+        sim_params = {}
+        sim = create_sim()
+        sim.run_sim(**sim_params)
+else:
+    def test_run_sim():
+        sim_params = {
+                "diffuse_foreground": {"Tsky_mdl":HERA_Tsky_mdl['xx']},
+                "pntsrc_foreground": {"nsrcs":500, "Smin":0.1},
+                "noiselike_eor": {"eor_amp":3e-2},
+                "thermal_noise": {"Tsky_mdl":HERA_Tsky_mdl['xx'], "inttime":8.59},
+                "rfi_scatter": {"chance":0.99, "strength":5.7, "std":2.2},
+                "rfi_impulse": {"chance":0.99, "strength":17.22},
+                "rfi_stations": {},
+                "gains": {"gain_spread":0.05},
+                "sigchain_reflections": {"amp":[0.5,0.5],
+                                         "dly":[14,7],
+                                         "phs":[0.7723,3.2243]},
+                "gen_whitenoise_xtalk": {"amplitude":1.2345} 
+                }
+
+        sim = create_sim()
+    
+        sim.run_sim(**sim_params)
+
+        assert not np.all(np.isclose(sim.data.data_array, 0))
+
+        # instantiate a mock simulation file
+        tmp_sim_file = tempfile.mkstemp()[1]
+        # write something to it
+        with open(tmp_sim_file, 'w') as sim_file:
+            sim_file.write("""
+                diffuse_foreground: 
+                    Tsky_mdl: 
+                        file: {}/HERA_Tsky_Reformatted.npz
+                        pol: yy
+                pntsrc_foreground: 
+                    nsrcs: !!int 500
+                    Smin: !!float 0.1
+                noiselike_eor: 
+                    eor_amp: !!float 0.03
+                gains: 
+                    gain_spread: !!float 0.05
+                gen_whitenoise_xtalk: 
+                    amplitude: !!float 1.2345
+                thermal_noise: 
+                    Tsky_mdl: 
+                        file: {}/HERA_Tsky_Reformatted.npz
+                        pol: xx
+                    inttime: !!float 9.72
+                rfi_scatter: 
+                    chance: !!float 0.99
+                    strength: !!float 5.7
+                    std: !!float 2.2
+                    """.format(DATA_PATH, DATA_PATH))
+        sim = create_sim()
+        sim.run_sim(tmp_sim_file)
+        assert not np.all(np.isclose(sim.data.data_array, 0))
+        # delete the temp file
+        remove(tmp_sim_file)
```

Comment:
I don't think you should have to manually remove a temp file?

Reply:
For these tests, I created the temporary file using `tempfile.mkstemp`, which requires the user to handle data cleanup, so that files need to be manually removed. I can figure out how to work with the `tempfile.NamedTemporaryFile` class so that `tempfile` handles cleanup automatically if that's what you'd prefer.

Reply:
Nope, all good.